<h1>ACTIVITY 1: PALMER PENGUINS DATA ANALYTICS PROJECT </h1>
<hr>

<h4>ANALYST: JUSTINE DESOACIDO</h4>

In [51]:
#libraries

In [52]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')
import matplotlib.pyplot as plt

In [53]:
bird = pd.read_csv ('datasets\\penguins.csv')

In [54]:
bird

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h1> DESCRIPTIVE ANALYTICS </h1>

In [55]:
bird.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [56]:
bird.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h1>FILTERING A COLUMN/DATA</h1>

In [57]:
bird['species']

0         Adelie
1         Adelie
2         Adelie
3         Adelie
4         Adelie
         ...    
339    Chinstrap
340    Chinstrap
341    Chinstrap
342    Chinstrap
343    Chinstrap
Name: species, Length: 344, dtype: object

In [58]:
adelie_df = bird[bird["species"]== 'Adelie']

In [59]:
adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [60]:
adelie_df['body_mass_g'].mean()

3700.662251655629

In [61]:
adelie_df['species'].value_counts()

species
Adelie    152
Name: count, dtype: int64

In [62]:
adelie_dream_df = bird[(bird ['species'] == 'Adelie') & (bird['island'] == 'Dream')]

<h1> CLEANING THE DATA FRAME</h1>

In [63]:
bird.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h1> WHAT TO DO WITH ALL THE "NULL VALUES" </h1>


In [64]:
bird['bill_length_mm'].mean()

43.9219298245614

In [65]:
bird['sex'].mode()


0    male
Name: sex, dtype: object

<h1> HOW TO FILL-UP NULL VALUES </h1>

In [66]:
bird['bill_length_mm'] = bird['bill_length_mm'].fillna(bird['bill_length_mm'].mean())
bird['bill_depth_mm'] = bird['bill_depth_mm'].fillna(bird['bill_depth_mm'].mean())
bird['flipper_length_mm'] = bird['flipper_length_mm'].fillna(bird['flipper_length_mm'].mean())
bird['body_mass_g'] = bird['body_mass_g'].fillna(bird['body_mass_g'].mean())

In [67]:
bird['sex'] = bird['sex'].fillna(bird['sex'].mode()[0])

In [68]:
bird.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h1> CHANGING DATA TYPES </h1>

In [69]:
bird['species'] = bird['species'].astype('category')
bird['year'] = bird['year'].astype('category')
bird['island'] = bird['island'].astype('category')
bird['sex'] = bird['sex'].astype('category')

In [70]:
bird.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


<h1> EXPLORATORY DATA ANALYSIS(EDA) </h1>

In [71]:
bird['body_mass_g'].mean()

4201.754385964912

In [72]:
bird['species'] == 'Adelie'

0       True
1       True
2       True
3       True
4       True
       ...  
339    False
340    False
341    False
342    False
343    False
Name: species, Length: 344, dtype: bool

In [73]:
adelie_df = bird['body_mass_g']

In [74]:
adelie_df.mean()

4201.754385964912

In [75]:
bird[bird['species'] == 'Adelie']['body_mass_g'].mean()

3703.95891043398

<h3>1. What is the maximum 'flipper_length_mm' recorded for Adelie penguins?</h3>

In [76]:
max_flipper_adelie = bird[bird ['species'] == 'Adelie']['flipper_length_mm'].max()

In [77]:
max_flipper_adelie

210.0

<h3>Insights</h3>
<h4>The maximum 'flipper_length_mm' recorded for Adelie penguins is 210.0</h4>

<h3>2. Is there a correlation between bill length and bill depth? </h3>

In [78]:
corr = np.corrcoef(bird["bill_length_mm"].dropna(), bird["bill_depth_mm"].dropna()) [0,1]

In [79]:
corr

-0.23505287035553282

<h3>Insights</h3>
<h4>correlation between bill length and bill depth is 0.23505287035553282 </h4>

<h3>3. What is the range of bill depth(mm) values for Adelie penguins?</h3>

In [80]:
bill_depth_range_adelie = bird[bird['species'] == 'Adelie']['bill_depth_mm'].max() - bird[bird['species'] == 'Adelie']['bill_depth_mm'].min() 

In [81]:
bill_depth_range_adelie 

6.0

<h3>Insights</h3>
<h4>The range of bill depth mm for Adelie 6.0 </h4>

<h3>4. How does the 'mean_body_mass_g' differ between male and female chinstrap peguins? </h3>

In [82]:
mean_bodymass_chinstrap = bird[bird['species'] == 'Chinstrap'].groupby('sex') ['body_mass_g'].mean()

In [83]:
mean_bodymass_chinstrap

sex
female    3527.205882
male      3938.970588
Name: body_mass_g, dtype: float64

<h3>Insights</h3>
<h4>Male chinstrap penguins has  3938.970588 mean body mass g </h4>
<h4>Female chinstrap penguins has 3527.205882 mean body mass g </h4>

<h3>5. What is the correlation between flipper length and year? </h3>

In [84]:
corr_flipper_year = bird['flipper_length_mm'].corr(bird['year'])

In [85]:
corr_flipper_year 

0.16893420971651854

<h3>Insights</h3>
<h4>The correlation between flipper length and year is 0.16893420971651854 </h4>

<h3>6. How many penguins have bill length greater than 50 mm?</h3>

In [86]:
count_bill_gt50 = bird[bird['bill_length_mm'] > 50].shape[0]


In [87]:
count_bill_gt50

52

<h3>Insights</h3>
<h4>There are 52 penguins has the bill length greater than 50mm</h4>

<h3>7. Which sex (male or female) has the highest average body mass ?</h3>

In [88]:
id = bird.groupby("sex")['body_mass_g'].mean().idxmax()

In [89]:
id

'male'

<h3>Insights</h3>
<h4>Male has the highest average body mass</h4>

<h3>8. What is the avergae bill depth for penguins from biscoe island?</h3>

In [90]:
avg_bill_depth_biscoe = bird[bird['island'] == 'Biscoe']['bill_depth_mm'].mean()

In [91]:
avg_bill_depth_biscoe

15.882447438039543

<h3>Insights</h3>
<h4>The average bill depth for penguins from  biscoe island is 15.882447438039543 </h4>

<h3>9. What is the average bill length for each island?</h3>

In [92]:
avg_bill_length_by_island = bird.groupby("island")["bill_length_mm"].mean()

In [93]:
avg_bill_length_by_island

island
Biscoe       45.249535
Dream        44.167742
Torgersen    39.046576
Name: bill_length_mm, dtype: float64

<h3>Insights</h3>
<h4>The average bill length for each island are Biscoe: 45.249535, 
Drea:m 44.16774, 
Torgers:   39.0465 76</h4>

<h3>10. On which island do we find the smallest penguin? </h3>

In [94]:
island_smallest_penguin = bird.loc[bird['body_mass_g'].idxmin(),'island'] 

In [95]:
island_smallest_penguin

'Dream'

<h3>Insights</h3>
<h4>We find the smallest penguin on Dream Island</h4>

<h3>11. What is the average body mass of the penguins for each year?</h3>

In [96]:
avg_body_mass_year = bird.groupby("year")["body_mass_g"].mean()


In [97]:
avg_body_mass_year

year
2007    4125.243222
2008    4266.666667
2009    4210.222953
Name: body_mass_g, dtype: float64

<h3>Insights</h3>
<h4>The average body mass of the penguins for each year 2007: 4125.243222, 2008: 4266.666667, 2009: 4210.222953</h4>

<h3>12. What is the median flipper length for male penguins of the Adelie species?</h3>

In [98]:
adelie_male_flipper_median = bird[(bird['species'] == 'Adelie') & (bird['sex'] == 'male')]["flipper_length_mm"].median()


In [99]:
adelie_male_flipper_median

192.0

<h3>Insights</h3>
<h4>The median flipper length for male penguins of the Adelie species is 192 </h4>